backend/typescript/services/implementations/camperService.ts

```diff
@@ -37,20 +37,21 @@ class CamperService implements ICamperService {
       if (campers.length > 0) {
         try {
           newCamperIds = newCampers.map((newCamper) => newCamper.id);
-          existingCampSession = await MgCampSession.findByIdAndUpdate(
+          existingCampSession = await MgCampSession.findById(
             campers[0].campSession,
-            {
-              $push: { campers: newCamperIds },
-            },
-            {
-              runValidators: true,
-            },
           );
+
           if (!existingCampSession) {
             throw new Error(
               `Camp session ${campers[0].campSession} not found.`,
             );
           }
+
+          existingCampSession.campers = existingCampSession.campers
+            .map((id) => id.toString())
+            .concat(newCamperIds);
+          await existingCampSession.save();
```

Comment:
ok so. we had to change from using findByIdAndUpdate to this because validators only work properly on document saving

Reply:
for some reason converting from string to objectId was BROKEN. but I realized when you pass string values mongoose auto converts those strings to object ids. that's why I added string type to campers field in CampSession interface and I'm just passing strings.